### PR TITLE
Feature/master parser xray

### PR DESCRIPTION
**Pull request by renejal**
### Feature new paser xray

### CODE REVIEW

**Review by ajtortolero — DISMISSED** [review]
Aprobado

File: docs/content/en/integrations/parsers/file/jfrog_xray_vulnerabilities.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+title: "JFrog XRay Vulnerabilities"
+toc_hide: true
+---
+Import the JSON format for the \"Security Export Vulnerabilities\" file. Use this importer for Xray version 2.X
```
Comment (Gaviria9601) [inline docs/content/en/integrations/parsers/file/jfrog_xray_vulnerabilities.md]: Change description according to the documentation https://jfrog.com/help/r/jfrog-cli/on-demand-binary-scan

File: docs/content/en/integrations/parsers/file/jfrog_xray_vulnerabilities.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+title: "JFrog XRay Vulnerabilities"
```
Comment (Gaviria9601) [inline docs/content/en/integrations/parsers/file/jfrog_xray_vulnerabilities.md]: Change to Jfrog Xray On-Demand Binary Scan

File: dojo/fixtures/defect_dojo_sample_data.json [hunk 3]
```diff
@@ -8620,6 +8620,16 @@
     }
     },
     {
+    "model": "dojo.test_type",
+    "pk": 149,
+    "fields": {
+        "name": "JFrog Xray Vulnerabilities",
```
Comment (Gaviria9601) [inline dojo/fixtures/defect_dojo_sample_data.json:8626]: Change  to Jfrog Xray On-Demand Binary Scan

File: dojo/settings/settings.dist.py [hunk 4]
```diff
@@ -1344,6 +1344,7 @@ def saml2_attrib_map_format(dict):
     'Bugcrowd API Import': ['unique_id_from_tool'],
     'Rubocop Scan': ['vuln_id_from_tool', 'file_path', 'line'],
     'JFrog Xray Scan': ['title', 'description', 'component_name', 'component_version'],
+    'JFrog Xray': ['title', 'description', 'component_name', 'component_version'],
```
Comment (Gaviria9601) [inline dojo/settings/settings.dist.py]: Change Jfrog Xray On-Demand Binary Scan

File: dojo/tools/jfrog_xray_vulnerabilities/parser.py [hunk 5]
```diff
@@ -0,0 +1,215 @@
+import json
+import re
+
+from cvss import CVSS3
+
+from dojo.models import Finding
+
+
+class JFrogXrayVulnerabilitiesParser(object):
```
Comment (Gaviria9601) [inline dojo/tools/jfrog_xray_vulnerabilities/parser.py]: Change Jfrog Xray On-Demand Binary Scan

File: dojo/tools/jfrog_xray_vulnerabilities/parser.py [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+    """jfrog_xray_scan JSON reports"""
+
+    def get_scan_types(self):
+        return ["JFrog Xray Vulnerabilities"]
```
Comment (Gaviria9601) [inline dojo/tools/jfrog_xray_vulnerabilities/parser.py]: Change Jfrog Xray On-Demand Binary Scan

File: unittests/scans/jfrog_xray_vulnerabilities/many_vulns.json [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+          "severity": "High",
+          "components": {
+            "gav://org.yaml:snakeyaml:1.16": {
+              "fixed_versions": [
+                "[1.26]"
+              ],
+              "impact_paths": [
+                [
+                  {
+                    "component_id": "gav://co.com.bancolombia.hygieia:core:1.0.0-SNAPSHOT"
```
Comment (Gaviria9601) [inline unittests/scans/jfrog_xray_vulnerabilities/many_vulns.json]: delete this information type

File: unittests/scans/jfrog_xray_vulnerabilities/many_vulns.json [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+          "severity": "High",
+          "components": {
+            "gav://commons-beanutils:commons-beanutils:1.9.2": {
+              "fixed_versions": [
+                "[1.9.4]"
+              ],
+              "impact_paths": [
+                [
+                  {
+                    "component_id": "gav://co.com.bancolombia.hygieia:core:1.0.0-SNAPSHOT"
```
Comment (Gaviria9601) [inline unittests/scans/jfrog_xray_vulnerabilities/many_vulns.json]: delete this information type

File: unittests/scans/jfrog_xray_vulnerabilities/many_vulns.json [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+          "components": {
+            "gav://org.springframework.boot:spring-boot-autoconfigure:1.3.0.RELEASE": {
+              "fixed_versions": [
+                "[1.2.8.RELEASE]",
+                "[1.3.1.RELEASE]"
+              ],
+              "impact_paths": [
+                [
+                  {
+                    "component_id": "gav://co.com.bancolombia.hygieia:core:1.0.0-SNAPSHOT"
```
Comment (Gaviria9601) [inline unittests/scans/jfrog_xray_vulnerabilities/many_vulns.json]: delete this information type

File: unittests/scans/jfrog_xray_vulnerabilities/many_vulns.json [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+          "references": [
+            "http://spring.io/blog/2015/12/18/spring-boot-1-3-1-and-1-2-8-available-now",
+            "https://github.com/spring-projects/spring-boot/issues/4763",
+            "https://github.com/spring-projects/spring-boot/commit/edb16a13ee33e62b046730a47843cb5dc92054e6",
+            "https://issues.apache.org/jira/browse/KYLIN-1893",
+            "https://issues.apache.org/jira/browse/KYLIN-1867"
+          ]
+        }
+      ],
+      "component_id": "gav://co.com.bancolombia.hygieia:core:1.0.0-SNAPSHOT",
```
Comment (Gaviria9601) [inline unittests/scans/jfrog_xray_vulnerabilities/many_vulns.json]: delete this information type

File: unittests/scans/jfrog_xray_vulnerabilities/one_vuln.json [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+          "severity": "High",
+          "components": {
+            "gav://org.yaml:snakeyaml:1.16": {
+              "fixed_versions": [
+                "[1.26]"
+              ],
+              "impact_paths": [
+                [
+                  {
+                    "component_id": "gav://co.com.bancolombia.hygieia:core:1.0.0-SNAPSHOT"
```
Comment (Gaviria9601) [inline unittests/scans/jfrog_xray_vulnerabilities/one_vuln.json]: delete this information type

File: unittests/scans/jfrog_xray_vulnerabilities/one_vuln.json [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+          "references": [
+            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/PTVJC54XGX26UJVVYCXZ7D25X3R5T2G6/",
+            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/CKN7VGIKTYBCAKYBRG55QHXAY5UDZ7HA/",
+            "https://bitbucket.org/asomov/snakeyaml/issues/377/allow-configuration-for-preventing-billion",
+            "https://bitbucket.org/asomov/snakeyaml/wiki/Billion%20laughs%20attack",
+            "https://mvnrepository.com/artifact/org.yaml/snakeyaml/1.25/usages"
+          ]
+        }
+      ],
+      "component_id": "gav://co.com.bancolombia.hygieia:core:1.0.0-SNAPSHOT",
```
Comment (Gaviria9601) [inline unittests/scans/jfrog_xray_vulnerabilities/one_vuln.json]: delete this information type

File: unittests/tools/test_jfrog_xray_vulnerabilities_parser.py [hunk 13]
```diff
@@ -0,0 +1,33 @@
+from ..dojo_test_case import DojoTestCase
+from dojo.models import Test, Finding
+from dojo.tools.jfrog_xray_vulnerabilities.parser import \
+    JFrogXrayVulnerabilitiesParser, decode_cwe_number
+
+
+class TestJfrogJFrogXrayVulnerabilitiesParser(DojoTestCase):
```
Comment (Gaviria9601) [inline unittests/tools/test_jfrog_xray_vulnerabilities_parser.py]: Change to Jfrog Xray On-Demand Binary Scan

**Review by Gaviria9601 — COMMENTED** [review]
Please review the comments

File: unittests/scans/jfrog_xray_vulnerabilities/one_vuln.json (hunk 12 above)
Comment (renejal) [reply unittests/scans/jfrog_xray_vulnerabilities/one_vuln.json]: deleted

File: unittests/tools/test_jfrog_xray_vulnerabilities_parser.py (hunk 13 above)
Comment (renejal) [reply unittests/tools/test_jfrog_xray_vulnerabilities_parser.py]: deleted

File: unittests/scans/jfrog_xray_vulnerabilities/many_vulns.json (hunk 10 above)
Comment (renejal) [reply unittests/scans/jfrog_xray_vulnerabilities/many_vulns.json]: deleted

File: docs/content/en/integrations/parsers/file/jfrog_xray_vulnerabilities.md (hunk 1 above)
Comment (renejal) [reply docs/content/en/integrations/parsers/file/jfrog_xray_vulnerabilities.md]: rename

File: docs/content/en/integrations/parsers/file/jfrog_xray_vulnerabilities.md (hunk 2 above)
Comment (renejal) [reply docs/content/en/integrations/parsers/file/jfrog_xray_vulnerabilities.md]: rename

File: dojo/fixtures/defect_dojo_sample_data.json (hunk 3 above)
Comment (renejal) [reply dojo/fixtures/defect_dojo_sample_data.json:8626]: rename

File: dojo/settings/settings.dist.py (hunk 4 above)
Comment (renejal) [reply dojo/settings/settings.dist.py]: rename

**Comment by renejal** [comment]
all correted

File: docs/content/en/integrations/parsers/file/jfrog_xray_on_demand_binary_scan.md [hunk 14]
```diff
@@ -0,0 +1,9 @@
+---
+title: "Jfrog xray on demand binary Scan"
+toc_hide: true
+---
+Import the JSON format for the "\Jfrog xray on demand binary Scan\" file. Use this importer for Xray version 2.X
+-- 
+ jfron file documentation
+
+https://jfrog.com/help/r/jfrog-cli/downloading-updates-for-xray-s-database
```
Comment (Gaviria9601) [inline docs/content/en/integrations/parsers/file/jfrog_xray_on_demand_binary_scan.md]: Update https://jfrog.com/help/r/jfrog-cli/on-demand-binary-scan

File: dojo/tools/factory.py [hunk 15]
```diff
@@ -97,7 +97,10 @@ def requires_tool_type(scan_type):
 
 # iterate through the modules in the current package
 package_dir = str(Path(__file__).resolve().parent)
-for module_name in os.listdir(package_dir):
+lis  = os.listdir(package_dir)
+for module_name in lis:
+    if "jfrog_xray_on_demand_binary_scan" in module_name:
+        print("esta")
```
Comment (Gaviria9601) [inline dojo/tools/factory.py]: delete print

File: docs/content/en/integrations/parsers/file/jfrog_xray_on_demand_binary_scan.md (hunk 14 above)
Comment (renejal) [inline docs/content/en/integrations/parsers/file/jfrog_xray_on_demand_binary_scan.md]: correted

File: dojo/tools/factory.py (hunk 15 above)
Comment (renejal) [reply dojo/tools/factory.py]: deleted

**Review by Gaviria9601 — APPROVED** [review]
Good Job. Go ahead

**Pull request merged**